sql/sys_vars.inl

```diff
@@ -3063,3 +3064,120 @@ private:
                                  character_set_collations);
   }
 };
+
+
+class Sys_var_path: public sys_var
+{
+public:
+  Sys_var_path(const char *name_arg, const char *comment,
+               int flag_args, ptrdiff_t off, size_t size,
+               CMD_LINE getopt,
+               enum binlog_status_enum binlog_status_arg)
+   :sys_var(&all_sys_vars, name_arg, comment,
+            flag_args, off, getopt.id, getopt.arg_type,
+            SHOW_CHAR,
+            DEFAULT(0), nullptr, binlog_status_arg,
+            nullptr, nullptr, nullptr)
+  {
+    option.var_type|= GET_STR;
+  }
+
+private:
+
+  static bool st_from_item(THD *thd, Sql_path *path,
+                           Item *item)
+  {
+    String *value, buffer;
+    if (!(value= item->val_str_ascii(&buffer)))
+      return true;
+    return path->from_text(thd, value->charset(), value->to_lex_cstring());
+  }
+
+  static const uchar *make_value_ptr(THD *thd,
+                                     const Sql_path &path)
+  {
+    size_t nbytes= path.text_format_nbytes_needed();
+    char *buf= thd->alloc(nbytes + 1);
+    size_t length= path.print(buf, nbytes);
+    buf[length]= '\0';
+    return (uchar *) buf;
+  }
+
+private:
+  bool do_check(THD *thd, set_var *var) override
+  {
+    Sql_path *path= new (thd->alloc<Sql_path>(1)) Sql_path();
+    if (!path || st_from_item(thd, path, var->value))
+    {
+      path->free();
+      return true;
+    }
+
+    var->save_result.ptr= path;
+    return false;
+  }
+
+  void session_save_default(THD *thd, set_var *var) override
+  {
+    thd->variables.path.set(global_system_variables.path, 1);
+  }
+
+  void global_save_default(THD *thd, set_var *var) override
+  {
+    global_system_variables.path.init();
+  }
+
+  bool session_update(THD *thd, set_var *var) override
+  {
+    if (thd->spcont)
+    {
+      /*
+        Because we invalidate the sp caches, we can't set the session path
+        in a stored function.
```

Comment:
that's because you resolve names in the parser, if you'd do it at fix_fields it'd likely work